Sources/PNG/PNG.swift

```diff
@@ -131,16 +131,56 @@ extension PNG
             return (x ^ mask) + (mask & 1)
         }
 
-        let v:(Int16, Int16, Int16) = (.init(a), .init(b), .init(c))
```

Comment:
let’s tackle this one improvement at a time :) 

i am really surprised that *integer conversions* are a bottleneck here. have you raised this issue on the Swift Forums yet? to me this borders on a compiler bug, and i’d love to hear what some compiler engineers have to add here

Reply:
Nah I haven't raised an issue. I have a feeling that this is kinda the intended behaviour of debug mode (although in my opinion it should be doing more basic optimisations like this which are so trivial). In release mode it compiles to one instruction but in debug mode it doesn't specialise [the generic `Int16` initialiser (which takes a `T: BinaryInteger`)](https://developer.apple.com/documentation/swift/int16/init(_:)-409to), and ends up spending all its time reading generic metadata and stuff, which of course adds up to a lot of overhead when you're doing a lot of these generic operations per pixel (roughly).

The changes in this PR are kinda all just working around things that the compiler could pretty easily compile better in debug mode. The reason that I did it this way is that waiting for compiler changes is quite a long game. I'd be happy to start a discussion about debug mode performance on the Swift forums at some point but I'd also like to be able to have reasonable debug mode performance earlier than the compiler will evolve.

That reminds me, another toolchain improvement that I thought might help was if we had a way of specifying separate optimisation levels for specific dependencies of a target when building. That way I could just get users to always compile SwiftCrossUI with optimisations and then they can compile their app with whatever optimisation level they want and still have reasonable image loading times.